### ISSUE TITLE
Simplify repr function in SpatialData

### ISSUE THREAD

**Issue by LucaMarconato** [scverse/spatialdata#53]
Review from @kevinyamauchi 

> ```
> [spatialdata/_core/_spatialdata.py]
>         ##
>         descr = "SpatialData object with:"
>         for attr in ["images", "labels", "points", "polygons", "shapes", "table"]:
> ```
> nitpick: this function would likely be cleaner if we had a separate function for generating reprs for each element. We could then generate the repr for each element type and then concatenate them. This would also allow as to move a lot of the details off of the SpatialData class into util functions. I think we should strive to make the SpatialData class as simple/clean as possible since this will be most people's main contact with the library.
> 
> In general, I think long if/else blocks like this are hard to follow and hard to test.

**Comment by melonora** [comment]
Will close for now, no impact for user.
